### ISSUE TITLE
A bug in MultiAlignedData

### ISSUE THREAD

**Issue by gpengzhi** [asyml/texar-pytorch#172]
In `MultiAlignedData`, [here](https://github.com/asyml/texar-pytorch/blob/master/texar/torch/data/data/multi_aligned_data.py#L227), `data_name` is set to be `None` so that properies `text_name`, `text_id_name`, and `length_name` in `MonoTextData` class are set to be default values (`text`, `text_ids`, and `length` respectively). 

However, we never set `data_name` to be `None` for `ScalarData` in `MultiAlignedData` so that the property `data_name` is the set by the user.

If we have
```python
data = {
    "batch_size": 50,
    "datasets": [
        {
            "files": "./data/sst2.train.sentences.txt",  # text data
            "vocab_file": "./data/sst2.vocab",
            "data_name": "x"
        },
        {
            "files": "./data/sst2.train.labels.txt",  #scalar data
            "data_type": "int",
            "data_name": "y"
        }
    ]
}
```
As a result, [here](https://github.com/asyml/texar-pytorch/blob/master/texar/torch/data/data/multi_aligned_data.py#L492), a `KeyError` will be triggered because in `ScalarData`, `data_name` is `y` instead of `label`.

I only read through part of the `data` module. Please correct me if I am wrong. @huzecong
